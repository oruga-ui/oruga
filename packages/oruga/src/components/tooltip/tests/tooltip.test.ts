import { describe, test, expect, afterEach } from "vitest";
import { enableAutoUnmount, mount } from "@vue/test-utils";

import OTooltip from "@/components/tooltip/Tooltip.vue";

describe("OTooltip tests", () => {
    enableAutoUnmount(afterEach);

    test("render correctly", () => {
        const wrapper = mount(OTooltip);
        expect(!!wrapper.vm).toBeTruthy();
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.attributes("data-oruga")).toBe("tooltip");
        expect(wrapper.html()).toMatchSnapshot();
    });
});
