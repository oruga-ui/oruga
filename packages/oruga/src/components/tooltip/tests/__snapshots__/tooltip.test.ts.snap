// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OTooltip tests > render correctly 1`] = `
"<div class="o-tip" data-oruga="tooltip">
  <!--teleport start-->
  <transition-stub name="fade" appear="false" persisted="true" css="true">
    <div class="o-tip__content o-tip__content--auto" style="display: none;"><span class="o-tip__arrow o-tip__arrow--auto"></span>
      <!--
                        @slot Tooltip content, default is label prop
                    -->
    </div>
  </transition-stub>
  <!--teleport end-->
  <div class="o-tip__trigger" aria-haspopup="true">
    <!--
                @slot Tooltip trigger slot
                @binding {boolean} active - tooltip active state
            -->
  </div>
</div>"
`;
